### PR TITLE
Add documentation for Crawlera API keys and account upgrade

### PR DESCRIPTION
**Pull request by rdowinton**
Resolves RM #31205

@michalmo please review, thanks.

### CODE REVIEW

**Comment by michalmo** [comment]
Users can create one crawlera user per paid organization, any more will need to go through support. I think the docs should mention this to avoid confusion.

In the 'Using Crawlera with Scrapy' and 'Using Crawlera with Scrapy Cloud' sections it may be helpful to elaborate on CRAWLERA_USER and CRAWLERA_PASS values.

**Comment by rdowinton** [comment]
Updated PR.

File: crawlera.rst [hunk 1]
```diff
@@ -9,12 +9,23 @@ Crawlera is a HTTP/HTTPS downloader that routes your requests through a pool of
...
 ===========
 
-Testing Crawlera Credentials
-----------------------------
+Creating a Crawlera User
+------------------------
+
+You can create a new Crawlera user from the homepage by clicking the 'Add a crawlera user' button. A modal will pop up. Select the organization you wish to assign the user to and click 'Create'. Your Crawlera user will take the name of the organization you selected. You can only create one Crawlera user per organization, if you wish to create more than one please contact support.
+
+.. note:: If you have previously used a username and password to authenticate, you can continue to do so if you don't wish to use the API key. Please note, however, that this method of authentication is likely to be removed in the future.
```
Comment (pablohoffman) [inline crawlera.rst]: if "we are likely to remove" this method in the future we shouldn't say they should "continue to use it if they don't wish to use the API key", instead we should encourage them to migrate to API key.

**Comment by rdowinton** [comment]
Addressing @pablohoffman 's comment, I've now removed references to the old method of authenticating with Crawlera, so there are now only details on how to use the API key.

**Comment by michalmo** [comment]
@rdowinton, @pablohoffman is this ready to merge?

**Comment by rdowinton** [comment]
@michalmo I think so, if any other changes are necessary we can add them later.

**Pull request merged**